File: core/src/main/scala/pl/edu/agh/multiscalemodelling/processsimulation/naiveseedsgrowth/NaiveSeedsGrowthLogic.java
package pl.edu.agh.multiscalemodelling.processsimulation.naiveseedsgrowth;

import pl.edu.agh.multiscalemodelling.engine.logic.Logic;

public class NaiveSeedsGrowthLogic extends Logic {

    public NaiveSeedsGrowthLogic(int x, int y) {

        super();
        board = new NaiveSeedsGrowthBoard(x, y);

    }

    @Override
    public void click(int x, int y) {
        ((NaiveSeedsGrowthBoard) board).swap(x, y);
    }

}

File: core/src/main/scala/pl/edu/agh/multiscalemodelling/processsimulation/naiveseedsgrowth/NaiveSeedsGrowthLogic.scala
package pl.edu.agh.multiscalemodelling.processsimulation.naiveseedsgrowth

import pl.edu.agh.multiscalemodelling.engine.logic.Logic

class NaiveSeedsGrowthLogic(val x: Int, val y: Int) extends Logic {
  board = new NaiveSeedsGrowthBoard(x, y)

  override def click(x: Int, y: Int): Unit = board.asInstanceOf[NaiveSeedsGrowthBoard].swap(x, y)
}